# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/base/nsFrameTraversal.h
# Commit: 21cb609fe669
# Full Hash: 21cb609fe6690fbad6c6167e10ee2e7f9804a593
# Author: David Zbarsky <dzbarsky@gmail.com>
# Date: 2012-10-16 08:09:24
# Regressor Bug: 779684
# File Overlap Count: 1
# Description:
#   Backout bug 779684 for topcrash
# ==============================================================================

diff -r a96deba7de41 -r 21cb609fe669 layout/base/nsFrameTraversal.h
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/layout/base/nsFrameTraversal.h	Mon Oct 15 14:35:50 2012 -0400
@@ -0,0 +1,38 @@
+/* -*- Mode: C++; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+#ifndef NSFRAMETRAVERSAL_H
+#define NSFRAMETRAVERSAL_H
+
+#include "nsIFrame.h"
+#include "nsIFrameTraversal.h"
+
+nsresult NS_NewFrameTraversal(nsIFrameEnumerator **aEnumerator,
+                              nsPresContext* aPresContext,
+                              nsIFrame *aStart,
+                              nsIteratorType aType,
+                              bool aVisual,
+                              bool aLockInScrollView,
+                              bool aFollowOOFs);
+
+nsresult NS_CreateFrameTraversal(nsIFrameTraversal** aResult);
+
+class nsFrameTraversal : public nsIFrameTraversal
+{
+public:
+  nsFrameTraversal();
+  virtual ~nsFrameTraversal();
+
+  NS_DECL_ISUPPORTS
+
+  NS_IMETHOD NewFrameTraversal(nsIFrameEnumerator **aEnumerator,
+                               nsPresContext* aPresContext,
+                               nsIFrame *aStart,
+                               PRInt32 aType,
+                               bool aVisual,
+                               bool aLockInScrollView,
+                               bool aFollowOOFs);
+};
+
+#endif //NSFRAMETRAVERSAL_H